pkg/pgxconn/pgx_conn.go

```diff
@@ -19,35 +23,97 @@ type PgxBatch interface {
 type PgxConn interface {
 	Close()
 	Exec(ctx context.Context, sql string, arguments ...interface{}) (pgconn.CommandTag, error)
-	Query(ctx context.Context, sql string, args ...interface{}) (pgx.Rows, error)
+	Query(ctx context.Context, sql string, args ...interface{}) (PgxRows, error)
 	QueryRow(ctx context.Context, sql string, args ...interface{}) pgx.Row
 	CopyFrom(ctx context.Context, tableName pgx.Identifier, columnNames []string, rowSrc pgx.CopyFromSource) (int64, error)
 	CopyFromRows(rows [][]interface{}) pgx.CopyFromSource
 	NewBatch() PgxBatch
 	SendBatch(ctx context.Context, b PgxBatch) (pgx.BatchResults, error)
 }
 
+type PgxRows interface {
+	Next() bool
+	Scan(...interface{}) error
+	Err() error
+	Close()
+}
+
+// This is only used by querier to log the time consumed
+// by SQL queries executed using Query(), QueryRow()
+func NewQueryLoggingPgxConn(pool *pgxpool.Pool) PgxConn {
+	return &loggingConnImpl{
+		connImpl: connImpl{Conn: pool},
+		Conn:     pool,
+	}
+}
+
 func NewPgxConn(pool *pgxpool.Pool) PgxConn {
 	return &connImpl{
 		Conn: pool,
 	}
 }
 
+type loggingConnImpl struct {
+	connImpl
+	Conn *pgxpool.Pool
```

Comment:
you should need this because connImpl already has it you should be able to access via p.connImpl.Conn

Reply:
Ahh, Again missed the power of embedded types. :)